frontend/src/components/common/Filter/components/FilterMenu/index.tsx

```diff
@@ -78,32 +80,64 @@ export default function FilterMenu({
             />
           </InputGroupWrapper>
         ) : null}
+        {/* No matches */}
         {emptyItems ? (
           <NoMatches
             multiline /* required when no matches text length is longer than longest category value */
             shouldDismissPopover={false}
             text={"No matches found"}
           />
         ) : (
-          <MenuItemsScroller>
-            {menuItems.map(({ key, count, label, selected }) => (
-              <MenuItemWrapper key={key} isSelected={selected}>
-                <MenuItem
-                  icon={
-                    <SelectionIcon
-                      icon={selected ? IconNames.TICK : IconNames.BLANK}
-                    />
-                  }
-                  labelElement={count}
-                  onClick={() => onFilter(categoryKey, key)}
-                  shouldDismissPopover={!isMultiselect}
-                  text={label}
-                />
-              </MenuItemWrapper>
-            ))}
+          <MenuItemsScroller {...scrollerProps}>
+            {/* Pinned values */}
+            {filteredPinnedValues.length > 0 && (
+              <FilterMenuItems
+                categoryKey={categoryKey}
+                isMultiselect={isMultiselect}
+                menuItems={filteredPinnedValues}
+                onFilter={onFilter}
+              />
+            )}
+            {/* Menu divider */}
+            {isMenuDivided && <MenuDivider />}
+            {/* Unpinned values */}
+            <FilterMenuItems
+              categoryKey={categoryKey}
+              isMultiselect={isMultiselect}
+              menuItems={filteredUnpinnedValues}
+              onFilter={onFilter}
+            />
           </MenuItemsScroller>
         )}
       </Menu>
     </MenuWrapper>
   );
 }
+
+/**
+ * Returns filtered category values where category key includes search value.
+ * @param values
```

Comment:
discussion: If the params don't have special comments needed, maybe we can drop their JSDoc?

Reply:
Thanks @tihuan I'll fix this now.